class.two-factor-force.php

```diff
@@ -0,0 +1,488 @@
+<?php
+/**
+ * Class for handling 2fa forcing.
+ *
+ * @since 0.1-dev
+ *
+ * @package Two_Factor
+ */
+class Two_Factor_Force {
+	/**
+	 * The network forced 2fa user roles key.
+	 *
+	 * @type string
+	 */
+	const FORCED_ROLES_META_KEY = '_two_factor_forced_roles';
+
+	/**
+	 * The network forced 2fa global key.
+	 *
+	 * @type string
+	 */
+	const FORCED_SITE_META_KEY = '_two_factor_forced_universally';
+
+	/**
+	 * Set up filters and actions.
+	 *
+	 * @since 0.1-dev
+	 */
+	public static function add_hooks() {
+		// Forced 2fa login functionality.
+		add_action( 'init', array( __CLASS__, 'register_scripts' ) );
+		add_action( 'wp_ajax_two_factor_force_form_submit', array( __CLASS__, 'handle_force_2fa_submission' ) );
+
+		// Handling intercession in 2 separate hooks to allow us to properly parse for REST requests.
+		add_filter( 'login_redirect', array( __CLASS__, 'maybe_redirect_login' ), 15, 3 );
+		add_action( 'parse_request', array( __CLASS__, 'maybe_redirect_to_2fa_settings' ) );
+		add_action( 'admin_init', array( __CLASS__, 'maybe_redirect_to_2fa_settings' ) );
+		add_action( 'admin_init', array( __CLASS__, 'maybe_display_2fa_settings' ) );
+
+		if ( is_multisite() ) {
+			add_action( 'wpmu_options', array( __CLASS__, 'force_two_factor_setting_options' ) );
+			add_action( 'update_wpmu_options', array( __CLASS__, 'save_network_force_two_factor_update' ) );
+		} else {
+			add_action( 'admin_init', array( __CLASS__, 'register_single_site_force_2fa_options' ) );
+		}
+	}
+
+	/**
+	 * Register scripts.
+	 */
+	public static function register_scripts() {
+		// Script for handling AJAX submission in force 2fa takeover screen.
+		wp_register_script(
+			'two-factor-form-script',
+			plugins_url( 'assets/js/force-2fa.js', __FILE__ ),
+			[],
+			'0.1',
+			false
```

Comment:
Could we add this to the footer instead?

Reply:
Yes, we probably could to that 👌